### PR TITLE
Adding new code for date formatter

### PR DESCRIPTION
**Pull request by jaywoo123**
First attempt at implementing a date formatter that uses unique() or sum() to find out date granularity.

### CODE REVIEW

**Comment by dorisjlee** [comment]
Thanks @jaywoo123! This looks good overall, I left some review comments in the code. Can we also add test cases to make sure this is working for a few of the datasets that you've looked at?

File: lux/utils/utils.py [hunk 1]
```diff
@@ -25,4 +27,34 @@ def getFilterSpecs(specLst):
...
+	elif granularity == "day":
+		dateStr += str(datetime.year) +"-"+ str(datetime.month) +"-"+ str(datetime.day)
+	else:
+		# non supported granularity
+		return datetime.date()
+
+	return dateStr
+
+
+def computeDateGranularity(ldf):
```
[the remark this answers is not in the document]
Comment (jaywoo123) [reply lux/utils/utils.py]: Added doc string to date related functions

File: lux/luxDataFrame/LuxDataframe.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def __init__(self,*args, **kw):
         self.executor = PandasExecutor
         self.SQLconnection = ""
         self.table_name = ""
+        self.dateGranularity = ""
```
[the remark this answers is not in the document]
Comment (jaywoo123) [reply lux/luxDataFrame/LuxDataframe.py]: changed code to remove granularity to be generated on demand.

**Comment by dorisjlee** [comment]
This looks great, thanks @jaywoo123 !

**Pull request merged**